baremetal/metal3machine_manager.go

```diff
@@ -150,6 +153,68 @@ func (m *MachineManager) role() string {
 	return bmRoleNode
 }
 
+// CheckPauseAnnotation checks and/or Removes the pause annotations on associated bmh
+func (m *MachineManager) CheckPauseAnnotation(ctx context.Context) error {
+	// look for associated BMH
+	host, err := m.getHost(ctx)
+	if err != nil {
+		m.setError("Failed to get a BaremetalHost for the Metal3Machine",
+			capierrors.CreateMachineError,
+		)
+		return err
+	}
+
+	if host == nil {
+		return nil
+	}
+
+	annotations := host.GetAnnotations()
+
+	if annotations != nil {
+		if _, ok := annotations[bmh.PausedAnnotation]; ok {
+			if m.Cluster.Name == host.Labels[capi.ClusterLabelName] && annotations[bmh.PausedAnnotation] == pausedAnnotationKey {
+				// Removing BMH Paused Annotation Since Owner Cluster is not paused
+				delete(host.Annotations, bmh.PausedAnnotation)
+			} else if m.Cluster.Name == host.Labels[capi.ClusterLabelName] && annotations[bmh.PausedAnnotation] != pausedAnnotationKey {
+				m.Log.Info("BMH is paused by user. Not removing Pause Annotation")
+				return nil
+			}
+		}
+	}
+	return m.updateObject(ctx, host)
```

Comment:
We should probably handle this kind of function differently. I'll create an issue so that we call the updateObject on the BMH only once at the end of the reconciliation. Otherwise we call it multiple time and make ourselves prone to conflicts because BMO will reconcile in the meantime.